FEXCore/Source/Interface/Core/OpcodeDispatcher/Flags.cpp

```diff
@@ -162,27 +160,35 @@ OrderedNode *OpDispatchBuilder::GetPackedRFLAG(uint32_t FlagsMask) {
 
     if ((GetNZ && (FlagOffset == FEXCore::X86State::RFLAG_SF_LOC ||
                    FlagOffset == FEXCore::X86State::RFLAG_ZF_LOC)) ||
-        FlagOffset == FEXCore::X86State::RFLAG_CF_LOC) {
+        FlagOffset == FEXCore::X86State::RFLAG_CF_LOC ||
+        FlagOffset == FEXCore::X86State::RFLAG_PF_LOC) {
       // Already handled
       continue;
     }
 
     // Note that the Bfi only considers the bottom bit of the flag, the rest of
     // the byte is allowed to be garbage.
     OrderedNode *Flag;
-    if (FlagOffset == FEXCore::X86State::RFLAG_PF_LOC)
-      Flag = LoadPF();
-    else if (FlagOffset == FEXCore::X86State::RFLAG_AF_LOC)
+    if (FlagOffset == FEXCore::X86State::RFLAG_AF_LOC)
       Flag = LoadAF();
     else
       Flag = GetRFLAG(FlagOffset);
 
-    if (Nonzero)
-      Original = _Orlshl(OpSize::i64Bit, Original, Flag, FlagOffset);
-    else
-      Original = _Lshl(OpSize::i64Bit, Flag, _Constant(FlagOffset));
+    Original = _Orlshl(OpSize::i64Bit, Original, Flag, FlagOffset);
+  }
 
-    Nonzero = true;
+  // Raw PF value needs to have its bottom bit masked out and inverted. The
+  // naive sequence is and/eor/orlshl. But we can do the inversion implicitly
+  // instead.
+  if (FlagsMask & (1 << FEXCore::X86State::RFLAG_PF_LOC)) {
+    // Set every bit except the bottommost.
+    auto OnesInvPF = _Or(OpSize::i64Bit, LoadPFRaw(), _Constant(~(uint64_t)1));
```

Comment:
You can use integer literal suffixes for this to make this nicer. `ULL` suffix is what we landed on in the codebase for uin64_t.
eg: `~1ULL`

Reply:
wasn't sure about the portability

Reply:
We require C++20 everywhere and this is in the spec since...C++11 I think? Used all over the place so this wouldn't be the first :)

Reply:
changed regardless